examples/speculative/speculative.cpp

```diff
@@ -37,12 +38,17 @@ int main(int argc, char ** argv) {
     // max number of parallel drafting sequences (i.e. tree branches)
     const int n_seq_dft = params.n_parallel;
 
-    // probability threshold for accepting a token from the draft model
-    const float p_accept = params.p_accept;
-
     // probability threshold for splitting a draft branch (only for n_seq_dft > 1)
     const float p_split  = params.p_split;
 
+    std::mt19937 r_gen;
+    if (params.seed >= 0) {
+        r_gen = std::mt19937(params.seed);
+    } else {
+        r_gen = std::mt19937(time(NULL));
+    }
+    std::uniform_int_distribution<std::mt19937::result_type> u_dist(0, RAND_MAX);
```

Comment:
This would be more idiomatic and consistent with the other examples:
```suggestion
    if (params.seed == LLAMA_DEFAULT_SEED) {
        params.seed = time(NULL);
    }
    std::default_random_engine r_gen(params.seed);
    std::uniform_real_distribution<> u_dist;
```

And then don't divide by RAND_MAX later.